macq/extract/esam.py

```diff
@@ -25,11 +27,14 @@ def make_PHashFluent_set(action: Action, flu: Fluent, action_2_sort: dict[str, l
     action = act(o1,o1)
     flu= lit(O1)
     output = {('lit', [object], [1]), ('lit', [object], [2])}
+
+    Parameters
+    ----------
     """
     ret: set[PHashLearnedLiftedFluent] = set()
     all_act_inds: list[list[int]] = list(map(list,
                                              product(*find_indexes_in_l2(flu.objects, action.obj_params))))
-    sorts: list[str] = [action_2_sort[action.name][i] for i in all_act_inds[0]]
+    sorts: list[str] = [action_2_sort[action.name][i] for i in all_act_inds[0]] if fluent_types is None else fluent_types[flu.name]
     if all(len(act_inds) > 0 for act_inds in all_act_inds):
         for act_inds in all_act_inds:  # for each product add the matching fluent information to the dict
             ret.add(PHashLearnedLiftedFluent(flu.name, sorts, list(act_inds)))
```

Comment:
Is this class part of your code? if so, add it to the review.
If it's not, add more informative comment cause I don't understand what this class is...